features/storage/csi_clone.feature

```diff
@@ -0,0 +1,67 @@
+Feature: CSI clone testing related feature
+  """
+  This test can only work on a Cluster which has CSI driver deployed, OSP has csi deploy by default from 4.7. 
+  When deploy a cluster which has no csi deployed, please use enable_tpnu_featureset: "yes" in LAUNCHER_VARS.
+  From 4.7, OSP CSI driver was installed by default. Its storage class name is standard-csi.
+  Use below command can temporarily add cs class(oc get sc, can find all deployed stroage class):
+  oc patch featuregate cluster -p '{"spec": {"featureSet": "TechPreviewNoUpgrade"}}' --type merge
+  """
+  
+  # @author jianl@redhat.com
+  # @case_id OCP-27615
+  @admin
+  Scenario: Clone a pvc on OSP in the same namespace is successfully
+    Given the master version >= "4.8"
+    Given I switch to cluster admin pseudo user
+    # Step 1
+    # From 4.7, OSP CSI driver was installed by default. The storage class name is standard-csi
+    # Step 2
+    When admin creates a project with:
+      | project_name | clone |
+    Then the step should succeed
+    Given I use the "clone" project
+    When I obtain test data file "storage/csi/pod_pvc_sc.yaml"
+    And I run the :create client command with:
+      | f | pod_pvc_sc.yaml |
+    Then the step should succeed
+    And the pod named "mypod" becomes ready
+    And the "mypvc" PVC becomes :bound
+    # Step 3
+    Given 1 pod becomes ready with labels:
+      | name=frontendhttp |
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | touch /mnt/local/testfile |
+    Then the step should succeed
+    """
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | echo "clone test" > /mnt/local/testfile |
+    Then the step should succeed
+    """
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | more /mnt/local/testfile |
+    Then the step should succeed
+    """
+    And the output should contain "clone test"
+    # Step 4
+    When I obtain test data file "storage/csi/pod_clone.yaml"
+    And I run the :create client command with:
+      | f | pod_clone.yaml |
+    Then the step should succeed
+    And the pod named "mypod-clone" becomes ready
+    And the "mypvc-clone" PVC becomes :bound
+    # Step 5
+    Given 1 pod becomes ready with labels:
+      | name=frontendhttp-clone |
+    And I wait for the steps to pass:
+    """
+    When I execute on the pod:
+      | sh | -c | more /mnt/local/testfile |
+    Then the step should succeed
+    """
+    And the output should contain "clone test"
```

Comment:
How about add another step to write another file in this cloned pvc? That's will cover more for the clone test.

Reply:
Here I mean add one check point to write something new into the cloned pvc, not just check the existing file.

Reply:
Sure